OpenTelemetry (S3M) Collector Docs and Getting Started revisions (#1872)
SumoLogic/sumologic-documentation: docs/send-data/opentelemetry-collector/install-collector-on-linux.md

   ```shell
   $ sudo cat /etc/systemd/system/otelcol-sumo.service
   [Service]
   ...
   EnvironmentFile=-/etc/otelcol-sumo/env/*.env
   ```

1. Ensure that the `/etc/otelcol-sumo/env` directory exists:

   ```bash
   sudo mkdir -p /etc/otelcol-sumo/env
   ```

1. Create `/etc/otelcol-sumo/env/token.env` with your installation token, eg:

   ```text
   SUMOLOGIC_INSTALLATION_TOKEN=<your token>
   ```

   We use `SUMOLOGIC_INSTALLATION_TOKEN` in example, as it will be autoamtically used by recommended configuration.

1. Ensure that file has correct owner and permissions:

   ```bash
   sudo chmod 440 /etc/otelcol-sumo/env/token.env
   sudo chown otelcol-sumo:otelcol-sumo /etc/otelcol-sumo/env/token.env
   ```

1. Remove `install_token` overrides from `/etc/otelcol-sumo/conf.d/*.yaml`.
   You can find them using the following command:

   ```shell
   $ sudo grep -Rn install_token /etc/otelcol-sumo/conf.d
   /etc/otelcol-sumo/conf.d/common.yaml:3:    install_token: <some token>
   ```

1. Restart `otelcol-sumo` service:

   ```bash
   sudo systemctl restart otelcol-sumo
   ```

#### Running Binary Manually

> If your system does not support Systemd, or you don't want to crate a service, you can run collector manually.

Author: ```suggestion
If your system does not support Systemd, or you don't want to create a service, you can run the collector manually.
```

---
id: install-collector-on-linux
title: Installing Collector on Linux
sidebar_label: Install Collector on Linux

> description: Learn how to install Collector

Reviewer: ```suggestion
description: Learn how to install the Sumo Logic OpenTelemetry Collector.
```

---

import useBaseUrl from '@docusaurus/useBaseUrl';

Follow the steps in this topic to install or uninstall an OpenTelemetry Collector on Linux. See [OpenTelemetry Collector](/docs/send-data/opentelemetry-collector) for information on other operating systems.

## System Requirements​

The Sumo Logic OpenTelemetry Collector is supported on both amd64 and arm64 architectures.

Minimal resource requirements are the following:

* 200 MB of disk space
* 64 MB of RAM

## Install

You can install our OpenTelemetry Collector using either of the following methods:

* [Install script](#install-script)
* [Manual step-by-step installation](#manual-step-by-step-installation)

### Install Script

#### 1. Get the Installation token

Get your [installation token](https://help.sumologic.com/docs/manage/security/installation-tokens) if you don't have it already and assign it to an environment variable:

```bash
export SUMOLOGIC_INSTALLATION_TOKEN=<TOKEN>
```

#### 2. Run the Installation script

You can run the script in two ways:

> * By piping `curl` straight into `bash`:
> 
>    ```bash
>    sudo curl -Ls https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh | sudo -E bash -s --
>    ```
> 
> * By first downloading the script, inspecting its contents for security, and then running it:
> 
>    ```bash

Reviewer: eliminate line spaces

```suggestion
* By piping `curl` straight into `bash`:
   ```bash
   sudo curl -Ls https://github.com/SumoLogic/sumologic-otel-collector/releases/latest/download/install.sh | sudo -E bash -s --
   ```
* By first downloading the script, inspecting its contents for security, and then running it:
   ```bash
```

| `--yes`                     | `y`        | Disable confirmation asks.                                                                                                                                                   | no                         |
| `--uninstall`               | `u`        | Removes Sumo Logic Distribution for OpenTelemetry Collector from the system and disable Systemd service eventually. Use with `--purge` to remove all configurations as well. | no                         |
| `--purge`                   | `p`        | It has to be used with `--uninstall`. It removes all Sumo Logic Distribution for OpenTelemetry Collector related configuration and data.                                     | no                         |
| `--help`                    | `h`        | Prints help and usage.                                                                                                                                                       | no                         |

The following env variables can be used along with script:

| name                           | description        |

> |--------------------------------|--------------------|

Reviewer: ```suggestion
|:---------------------|:-----------------|
```

Move the downloaded binary into a directory from your `PATH` environment, so that it can be used by simply invoking `otelcol-sumo`.

```bash
chmod +x otelcol-sumo
sudo mv otelcol-sumo /usr/local/bin/otelcol-sumo
```

#### Step 3. Verify the Installation

In order to verify installation, run OpenTelemetry Collector.

```bash
otelcol-sumo --version
```

#### Step 4. Run OpenTelemetry Collector as Systemd Service

We recommend using the [installation script](#install-script) as it supports Systemd scenario.

This section describes how to install it manually.

:::note
Ensure that the user who will run the `otelcol-sumo` process has access to any directories used in your configuration within the filesystem.
:::

For example, if you use the [file_storage extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/storage/filestorage) in your configuration like this:

```yaml
 extensions:
 file_storage/custom_settings:
 directory: /var/lib/otelcol/mydir
 timeout: 1s
```

Then the user running the process must have access to `/var/lib/otelcol/mydir`.

To run Opentelemetry Collector as Systemd Service, follow the steps below:

> 1. Ensure that `otelcol-sumo` has been installed into `/usr/local/bin/otelcol-sumo` by running this command.
> 
>    ```bash
>    /usr/local/bin/otelcol-sumo --version
>    ```
> 
> 1. Create configuration, follow the steps below.
> 
>    1. Get [recommended configuration](https://github.com/SumoLogic/sumologic-otel-collector/blob/main/examples/sumologic.yaml) from Sumo Logic repository and save it as `/etc/otelcol-sumo/sumologic.yaml`.
>    1. Create your custom configuration file in the `/etc/otelcol-sumo/conf.d` directory. For example, you can create a file called `common.yaml` in this directory to store your custom configuration, like this `/etc/otelcol-sumo/conf.d/common.yaml`.

Reviewer: ```suggestion
1. Ensure that `otelcol-sumo` has been installed into `/usr/local/bin/otelcol-sumo` by running this command.
   ```bash
   /usr/local/bin/otelcol-sumo --version
   ```
1. Create configuration, follow the steps below.
   1. Get [recommended configuration](https://github.com/SumoLogic/sumologic-otel-collector/blob/main/examples/sumologic.yaml) from Sumo Logic repository and save it as `/etc/otelcol-sumo/sumologic.yaml`.
   1. Create your custom configuration file in the `/etc/otelcol-sumo/conf.d` directory. For example, you can create a file called `common.yaml` in this directory to store your custom configuration, like this `/etc/otelcol-sumo/conf.d/common.yaml`.
```

1. Ensure that the file has the correct owner and permissions by running this command.

   ```bash
   sudo chmod 440 /etc/otelcol-sumo/env/token.env
   sudo chown otelcol-sumo:otelcol-sumo /etc/otelcol-sumo/env/token.env
   ```

1. Remove `install_token` overrides from `/etc/otelcol-sumo/conf.d/*.yaml`. You can find them using the following command:

   ```shell
   $ sudo grep -Rn install_token /etc/otelcol-sumo/conf.d
   /etc/otelcol-sumo/conf.d/common.yaml:3:    install_token: <some token>
   ```

1. Restart `otelcol-sumo` service by running this command.

   ```bash
   sudo systemctl restart otelcol-sumo
   ```

#### Running Binary Manually

If your system does not support `Systemd`, or you don't want to create a service, you can run Collector manually.

```bash
sudo otelcol-sumo --config=/etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
```

### UI Installation via App Catalog

1. Go to **App Catalog** and select `Linux`.

<img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux.png')} alt="linux in app catalog" />

> 1. Select `Add New Collector` and click `Next`
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-collector.png')} alt="set up collector" />
> 
> 1. Select installation token and customise your tags.
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-register-collector.png')} alt="add new collector" />
> 
> 1. Copy command and execute it in your system terminal.
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/linux-terminal-installation.png')} alt="execute command in terminal" />
> 
> 1. Wait for THE installation to be completed.
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-registration-success.png')} alt="application installed successfully" />
> 
> 1. Read the prerequisite section.
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-prerequisite.png')} alt="collector successfully registered" />
> 
> 1. Customize source configuration, download it, and place it in the directory `/etc/otelcol-sumo/conf.d`, and then restart collector.
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-configure.png')} alt="source customisation" />
> 
> 1. Wait for the installation to be completed.
> 
>    <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-success.png')} alt="application installed successfully" />

Reviewer: ```suggestion
1. Select `Add New Collector` and click `Next`. <br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-collector.png')} alt="set up collector" />
1. Select installation token and customise your tags.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-register-collector.png')} alt="add new collector" />
1. Copy command and execute it in your system terminal.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/linux-terminal-installation.png')} alt="execute command in terminal" />
1. Wait for THE installation to be completed.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-registration-success.png')} alt="application installed successfully" />
1. Read the prerequisite section.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-prerequisite.png')} alt="collector successfully registered" />
1. Customize source configuration, download it, and place it in the directory `/etc/otelcol-sumo/conf.d`, and then restart collector.<br/><img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-configure.png')} alt="source customisation" />
1. Wait for the installation to be completed.<br/> <img src={useBaseUrl('img/send-data/opentelemetry-collector/app-catalog-linux-success.png')} alt="application installed successfully" />
```
